beacon_node/lighthouse_network/src/rpc/handler.rs

```diff
@@ -892,6 +892,20 @@ where
             ConnectionEvent::DialUpgradeError(DialUpgradeError { info, error }) => {
                 self.on_dial_upgrade_error(info, error)
             }
+            ConnectionEvent::ListenUpgradeError(e) => {
+                if matches!(e.error.1, RPCError::InvalidData(_)) {
```

Comment:
you don't need the nested matches Akihito, you can do:
```suggestion
            ConnectionEvent::ListenUpgradeError(ListenUpgradeError {
                error: (_, RPCError::IoError(_)),
                ..
            }) => {
```

Reply:
Refactored in https://github.com/sigp/lighthouse/pull/6986/commits/7e09c470c720129ce18a18177c0f43deab8040db